pkg/manager/internal.go

```diff
@@ -355,6 +359,19 @@ func (cm *controllerManager) serveHealthProbes() {
 	go cm.httpServe("health probe", cm.logger, server, cm.healthProbeListener)
 }
 
+func (cm *controllerManager) servePprof() {
```

Comment:
This should be a normal `Runnable` and not a method on the controllerManager

Reply:
also, don't forget to implement the runnable with `NeedLeaderElection() bool -> false` so it'll run also for non-leader instances

Reply:
> This should be a normal `Runnable` and not a method on the controllerManager

Sorry I don't quite get the point of it. As internal http servers, there're `serveMetrics` and `serveHealthProbes` methods in the manager, is there any difference between them and the pprof server?

Reply:
@alvaroaleman It would be grateful if you can give some further explanation on why it should be a normal `Runnable` rather than an internal http server like the metrics and health ones, thanks.

Reply:
Runnables can be closed and their lifecycle is pretty clear. Anything else outside runnables has the lifecycle outside the manager

Reply:
@vincepri But the http server started by `func (cm *controllerManager) httpServe` can also be closed within the manager's lifecycle. If using `Runnable` is a recommended way, why don't we also switch health probes and metrics server to it?
In other words, I'm not object to the `Runnable` way, but to the disunity within profiling server and health/metrics ones.

Reply:
The fact that they are not Runnables today is just historic, they should be (if you want to work on this, please make that a distinct, refactoring-only change)

Reply:
@alvaroaleman Thanks for the explanation, will do that after this PR get done.